nd4j/nd4j-serde/nd4j-arrow/src/main/java/org/nd4j/arrow/ByteDecoArrowSerde.java

```diff
@@ -0,0 +1,270 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Konduit KK
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+
+package org.nd4j.arrow;
+import org.bytedeco.arrow.global.arrow;
+import org.bytedeco.arrow.*;
+import org.nd4j.base.Preconditions;
+import org.nd4j.linalg.api.buffer.DataBuffer;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.factory.Nd4j;
+import org.nd4j.linalg.primitives.Pair;
+import org.nd4j.linalg.util.ArrayUtil;
+
+
+/**
+ *
+ */
+public class ByteDecoArrowSerde {
+
+    /**
+     * Convert a {@link Tensor}
+     * to an {@link INDArray}
+     * @param tensor the input tensor
+     * @return the equivalent {@link INDArray}
+     */
+    public static INDArray fromTensor(Tensor tensor) {
+        long[] shape = new long[tensor.ndim()];
+        long[] stride = new long[tensor.ndim()];
+
+        long bufferCapacity = 1;
+        for(int i = 0; i < tensor.ndim(); i++) {
+            shape[i] = tensor.shape().get(i);
+            stride[i] = tensor.strides().get(i);
+            bufferCapacity *= shape[i];
+        }
+
+
+        org.nd4j.linalg.api.buffer.DataType dtype = dataBufferTypeTypeForArrow(tensor.type());
+        //buffer capacity needs to be initialized properly, otherwise defaults to zero
+        ArrowBuffer arrowBuffer = tensor.data().capacity(bufferCapacity);
```

Comment:
The `capacity()` method from the C++ API gets mapped to `_capacity()`:
https://github.com/bytedeco/javacpp-presets/blob/master/arrow/src/gen/java/org/bytedeco/arrow/ArrowBuffer.java
But I think what we should use here is `size()`.

Reply:
@saudet  what's the difference? I'm not seeing a size() method I can plug in here.

Reply:
It's right there: https://github.com/bytedeco/javacpp-presets/blob/master/arrow/src/gen/java/org/bytedeco/arrow/ArrowBuffer.java#L149

Reply:
@saudet ah you're saying we should use size instead of capacity.  The problem comes when I pass that to the nd4j create buffer. That uses capacity not size.

Reply:
It uses the capacity of Pointer, not ArrowBuffer, so that's fine.

Reply:
@saudet mind reviewing my usage again? I fixed all of this.